tests/test_values.py

```diff
@@ -22,52 +30,63 @@
     ),
 )
 def test_literal_nodes(node, type):
-    resolved_value, resolved_type = ValueResolver({}).resolve(node)
+    resolved_value, resolved_type = ValueResolver(_make_context()).resolve(node)
     assert resolved_value == node.value
     assert resolved_type == type
 
 
 @pytest.mark.parametrize(("value", "unit"), ((1.2, "dt"), (4, "us")))
 def test_duration_literal(value, unit):
     node = ast.DurationLiteral(value, ast.TimeUnit[unit])
-    resolved_value, resolved_type = ValueResolver({}).resolve(node)
+    resolved_value, resolved_type = ValueResolver(_make_context()).resolve(node)
     assert resolved_value == (value, unit)
     assert resolved_type == types.Duration(const=True)
 
 
 def test_identifier():
-    symbols = {
-        "a": Symbol("a", 1, types.Int(const=True), Scope.GLOBAL),
-        "b": Symbol("b", True, types.Bool(const=True), Scope.GLOBAL),
-    }
-    resolver = ValueResolver(symbols)
-    for name, symbol in symbols.items():
+    symbols = [
+        Symbol("a", 1, types.Int(const=True), Scope.GLOBAL),
+        Symbol("b", True, types.Bool(const=True), Scope.GLOBAL),
+    ]
+    resolver = ValueResolver(_make_context(symbols))
+    for symbol in symbols:
+        name = symbol.name
         resolved_value, resolved_type = resolver.resolve(ast.Identifier(name=name))
         assert resolved_value == symbol.data
         assert resolved_type == symbol.type
     with pytest.raises(ConversionError, match="not defined in this scope"):
         resolver.resolve(ast.Identifier(name="c"))
 
 
+def test_physical_qubit_identifier():
+    resolver = ValueResolver(_make_context())
+    for name in ("$0", "$123"):
+        q0, q0_type = resolver.resolve(ast.Identifier(name=name))
+        assert isinstance(q0, Qubit)
+        assert isinstance(q0_type, types.HardwareQubit)
+    with pytest.raises(ConversionError, match="not defined in this scope"):
+        resolver.resolve(ast.Identifier(name="$q"))
```

Comment:
This isn't a valid OQ3 identifier, so we probably don't need to make any particular assertions about what happens with it.

Reply:
I agree "not defined in scope" is not correct, but I often like to test the behavior of the code as it is, even though it might be improved. (e.g. test that wrong results happen, or that a test is broken in the expected way). ... But, I agree it seems to be testing something that's outside the scope of this converter. I don't recall why I put it here.

Reply:
I'm fine to leave it.

`qss-qasm` accepts `$q` because historically that was allowed as a "wildcard" identifier for a hardware qubit in a `defcal` statement, but that was changed in openqasm/openqasm#371.

Reply:
Remove test involving invalid identifier `$q` done in 94fb45f50446

Reply:
> I'm fine to leave it.
I removed it before I saw this. Maybe I didn't refresh. But I'm ok with removing it.